Inf1: optimum neuron inference tests failing with import errors from transformers.utils and huggingface_hub

### System Info

```shell
Platform : U20 INF1
Instance type:  Inf1.6xlarge

Versions:
python 3.8


optimum-cli env

Copy-and-paste the text below in your GitHub issue:


Platform:

- Platform: Linux-5.15.0-1057-aws-x86_64-with-glibc2.29
- Python version: 3.8.10


Python packages:

- `optimum-neuron` version: 0.0.21
- `neuron-sdk` version: 2.18.0
- `optimum` version: 1.18.1
- `transformers` version: 4.36.2
- `huggingface_hub` version: 0.22.2
- `torch` version: 1.13.1+cu117
- `dmlc-tvm` version: 1.19.0.0+0
- `neuron-cc` version: 1.22.0.0+d4b4f5311
- `torch-neuron` version: 1.13.1.2.9.74.0


Neuron Driver:


WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

aws-neuronx-collectives/now 2.20.22.0-c101c322e amd64 [installed,local]
aws-neuronx-devtools/now 2.17.1.0 amd64 [installed,local]
aws-neuronx-dkms/now 2.16.7.0 amd64 [installed,local]
aws-neuronx-oci-hook/now 2.3.0.0 amd64 [installed,local]
aws-neuronx-runtime-lib/now 2.20.22.0-1b3ca6425 amd64 [installed,local]
aws-neuronx-tools/now 2.17.1.0 amd64 [installed,local]
```

(aws_neuron_venv) ubuntu@ip-10-4-210-72:~/optimum-neuron$ pip list | grep "parameterized"
parameterized            0.9.0
```


### Who can help?

@dacorvo, @JingyaHuang

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

Reproducible steps:
Launch Inf1.6xlarge instance
Install latest neuorn bits.

Use v0.0.21 version of the release for optimum-neuron.
pip install optimum-neuron
git clone --depth 1 --branch v0.0.21 https://github.com/huggingface/optimum-neuron.git
cd optimum-neuron/examples/translation
pip install -r requirements.txt
pytest -m is_inferentia_test tests

Error log:
```
================================================================================ test session starts =================================================================================
platform linux -- Python 3.8.10, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/ubuntu/optimum-neuron
configfile: pyproject.toml
plugins: remotedata-0.4.1, xdist-3.5.0, timeout-2.3.1
collected 0 items / 19 errors                                                                                                                                                        

======================================================================================= ERRORS =======================================================================================
_________________________________________________________________ ERROR collecting tests/cache/test_neuronx_cache.py _________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/cache/test_neuronx_cache.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/cache/test_neuronx_cache.py:29: in <module>
    from optimum.neuron import (
optimum/neuron/__init__.py:18: in <module>
    from transformers.utils import _LazyModule
E   ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
___________________________________________________________________ ERROR collecting tests/cli/test_export_cli.py ____________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/cli/test_export_cli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/cli/test_export_cli.py:20: in <module>
    from optimum.exporters.neuron.model_configs import *  # noqa: F403
optimum/exporters/neuron/__init__.py:17: in <module>
    from transformers.utils import _LazyModule
E   ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
_______________________________________________________________ ERROR collecting tests/cli/test_export_decoder_cli.py ________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/cli/test_export_decoder_cli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/cli/test_export_decoder_cli.py:7: in <module>
    from optimum.neuron.utils.testing_utils import is_inferentia_test, requires_neuronx
optimum/neuron/__init__.py:18: in <module>
    from transformers.utils import _LazyModule
E   ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
________________________________________________________________ ERROR collecting tests/cli/test_neuron_cache_cli.py _________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/cli/test_neuron_cache_cli.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/cli/test_neuron_cache_cli.py:24: in <module>
    from huggingface_hub import HfApi, create_repo, delete_repo
E   ImportError: cannot import name 'create_repo' from 'huggingface_hub' (/home/ubuntu/.local/lib/python3.8/site-packages/huggingface_hub/__init__.py)
_________________________________________________________________ ERROR collecting tests/distributed/test_common.py __________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/distributed/test_common.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/distributed/test_common.py:21: in <module>
    import safetensors
E   ModuleNotFoundError: No module named 'safetensors'
__________________________________________________________ ERROR collecting tests/distributed/test_model_parallelization.py __________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/distributed/test_model_parallelization.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/distributed/test_model_parallelization.py:23: in <module>
    from transformers import AutoTokenizer, LlamaForCausalLM
E   ImportError: cannot import name 'LlamaForCausalLM' from 'transformers' (unknown location)
________________________________________________________________ ERROR collecting tests/distributed/test_training.py _________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/distributed/test_training.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/distributed/test_training.py:21: in <module>
    from datasets import load_dataset
E   ModuleNotFoundError: No module named 'datasets'
__________________________________________________________________ ERROR collecting tests/distributed/test_utils.py __________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/distributed/test_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/distributed/test_utils.py:24: in <module>
    from safetensors.torch import save_file
E   ModuleNotFoundError: No module named 'safetensors'
__________________________________________________________________ ERROR collecting tests/exporters/test_export.py ___________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/exporters/test_export.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/exporters/test_export.py:24: in <module>
    from parameterized import parameterized
E   ModuleNotFoundError: No module named 'parameterized'
_________________________________________________________________________ ERROR collecting tests/generation __________________________________________________________________________
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
../.local/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
tests/generation/conftest.py:20: in <module>
    from optimum.neuron import NeuronModelForCausalLM, NeuronModelForSeq2SeqLM
optimum/neuron/__init__.py:18: in <module>
    from transformers.utils import _LazyModule
E   ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
_________________________________________________________________ ERROR collecting tests/inference/test_modeling.py __________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/inference/test_modeling.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/inference/test_modeling.py:23: in <module>
    from parameterized import parameterized
E   ModuleNotFoundError: No module named 'parameterized'
_________________________________________________________ ERROR collecting tests/inference/test_stable_diffusion_pipeline.py _________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/inference/test_stable_diffusion_pipeline.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/inference/test_stable_diffusion_pipeline.py:20: in <module>
    from parameterized import parameterized
E   ModuleNotFoundError: No module named 'parameterized'
__________________________________________________________________________ ERROR collecting tests/pipelines __________________________________________________________________________
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
../.local/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:178: in exec_module
    exec(co, module.__dict__)
tests/pipelines/conftest.py:20: in <module>
    from optimum.neuron import NeuronModelForCausalLM
optimum/neuron/__init__.py:18: in <module>
    from transformers.utils import _LazyModule
E   ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
_____________________________________________________________________ ERROR collecting tests/test_cache_utils.py _____________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/test_cache_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_cache_utils.py:26: in <module>
    from huggingface_hub import create_repo, delete_repo, get_token, login
E   ImportError: cannot import name 'create_repo' from 'huggingface_hub' (/home/ubuntu/.local/lib/python3.8/site-packages/huggingface_hub/__init__.py)
______________________________________________________________________ ERROR collecting tests/test_examples.py _______________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/test_examples.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_examples.py:27: in <module>
    from huggingface_hub import get_token
E   ImportError: cannot import name 'get_token' from 'huggingface_hub' (/home/ubuntu/.local/lib/python3.8/site-packages/huggingface_hub/__init__.py)
______________________________________________________________________ ERROR collecting tests/test_generate.py _______________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/test_generate.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_generate.py:6: in <module>
    from parameterized import parameterized
E   ModuleNotFoundError: No module named 'parameterized'
_______________________________________________________________________ ERROR collecting tests/test_runner.py ________________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/test_runner.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_runner.py:20: in <module>
    from huggingface_hub import get_token, login
E   ImportError: cannot import name 'get_token' from 'huggingface_hub' (/home/ubuntu/.local/lib/python3.8/site-packages/huggingface_hub/__init__.py)
______________________________________________________________________ ERROR collecting tests/test_trainers.py _______________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/test_trainers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_trainers.py:28: in <module>
    from huggingface_hub import HfApi, delete_repo
E   ImportError: cannot import name 'delete_repo' from 'huggingface_hub' (/home/ubuntu/.local/lib/python3.8/site-packages/huggingface_hub/__init__.py)
________________________________________________________________________ ERROR collecting tests/test_utils.py ________________________________________________________________________
ImportError while importing test module '/home/ubuntu/optimum-neuron/tests/test_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_utils.py:24: in <module>
    from optimum.neuron.accelerate.accelerator import MODEL_PATCHING_SPECS
optimum/neuron/__init__.py:18: in <module>
    from transformers.utils import _LazyModule
E   ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
============================================================================== short test summary info ===============================================================================
ERROR tests/cache/test_neuronx_cache.py
ERROR tests/cli/test_export_cli.py
ERROR tests/cli/test_export_decoder_cli.py
ERROR tests/cli/test_neuron_cache_cli.py
ERROR tests/distributed/test_common.py
ERROR tests/distributed/test_model_parallelization.py
ERROR tests/distributed/test_training.py
ERROR tests/distributed/test_utils.py
ERROR tests/exporters/test_export.py
ERROR tests/generation - ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
ERROR tests/inference/test_modeling.py
ERROR tests/inference/test_stable_diffusion_pipeline.py
ERROR tests/pipelines - ImportError: cannot import name '_LazyModule' from 'transformers.utils' (/home/ubuntu/.local/lib/python3.8/site-packages/transformers/utils/__init__.py)
ERROR tests/test_cache_utils.py
ERROR tests/test_examples.py
ERROR tests/test_generate.py
ERROR tests/test_runner.py
ERROR tests/test_trainers.py
ERROR tests/test_utils.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 19 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================= 19 errors in 1.90s ==============
```
```
[Tests in Optimum Neuron (3) (3).pdf](https://github.com/huggingface/optimum-neuron/files/15045870/Tests.in.Optimum.Neuron.3.3.pdf)
```










### Expected behavior

Expected to optimum-neuron tests to pass on Inf1.

Thanks for the quick reply.
I am not seeing previous error.
New error is as below.
```
repo_id = 'optimum-internal-testing/optimum-neuron-cache-for-testing'
default_cache = <libneuronxla.neuron_cc_cache.CompileCacheFs object at 0x7f3f6c0effd0>
endpoint = None, token = None

    def __init__(
        self, repo_id: str, default_cache: CompileCache, endpoint: Optional[str] = None, token: Optional[str] = None
    ):
        # Initialize the proxy cache as expected by the parent class
        super().__init__(default_cache.cache_url)
        self.cache_path = default_cache.cache_path
        # Initialize specific members
        self.default_cache = default_cache
        self.api = HfApi(endpoint=endpoint, token=token, library_name="optimum-neuron", library_version=__version__)
        # Check if the HF cache id is valid
        try:
            if not self.api.repo_exists(repo_id):
>               raise ValueError(f"The {repo_id} repository does not exist or you don't have access to it.")
E               ValueError: The optimum-internal-testing/optimum-neuron-cache-for-testing repository does not exist or you don't have access to it.

--- a/tests/inference/inference_utils.py
+++ b/tests/inference/inference_utils.py
@@ -13,6 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import shutil
 import tempfile
 import unittest
@@ -82,7 +83,10 @@ class NeuronModelIntegrationTestMixin(unittest.TestCase):
 
         # Upload to the hub
         cls.neuron_model_id = f"{cls.USER}/{cls.NEURON_MODEL_REPO}"
-        neuron_model.push_to_hub(model_dir, repository_id=cls.neuron_model_id, use_auth_token=cls._token)
+
+        use_auth_token = os.environ.get("HF_TOKEN", None)
+        if use_auth_token:
+            neuron_model.push_to_hub(model_dir, repository_id=cls.neuron_model_id, use_auth_token=cls._token)
 
     @classmethod
     def tearDownClass(cls):

--- a/tests/inference/inference_utils.py
+++ b/tests/inference/inference_utils.py
@@ -13,7 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import os
 import shutil
 import tempfile
 import unittest
@@ -84,8 +83,7 @@ class NeuronModelIntegrationTestMixin(unittest.TestCase):
         # Upload to the hub
         cls.neuron_model_id = f"{cls.USER}/{cls.NEURON_MODEL_REPO}"
 
-        use_auth_token = os.environ.get("HF_TOKEN", None)
-        if use_auth_token:
+        if cls._token:
             neuron_model.push_to_hub(model_dir, repository_id=cls.neuron_model_id, use_auth_token=cls._token)
 
     @classmethod

--- a/tests/inference/test_modeling.py
+++ b/tests/inference/test_modeling.py
@@ -52,7 +52,6 @@ from optimum.neuron.utils import NEURON_FILE_NAME, is_neuron_available, is_neuro
 from optimum.neuron.utils.testing_utils import is_inferentia_test, requires_neuronx
 from optimum.utils import CONFIG_NAME, logging
 
-from ..test_utils import require_hf_token
 from .inference_utils import (
     MODEL_NAMES,
     SEED,
@@ -66,7 +65,6 @@ logger = logging.get_logger()
 
 
 @is_inferentia_test
-@require_hf_token
 class NeuronModelIntegrationTest(NeuronModelIntegrationTestMixin):
     MODEL_ID = MODEL_NAMES["bert"]
     if is_neuron_available():

--- a/tests/inference/test_modeling.py
+++ b/tests/inference/test_modeling.py
@@ -50,11 +50,9 @@ from optimum.neuron import (
 )
 from optimum.neuron.utils import NEURON_FILE_NAME, is_neuron_available, is_neuronx_available
 from optimum.neuron.utils.testing_utils import is_inferentia_test, requires_neuronx
-from optimum.utils import (
-    CONFIG_NAME,
-    logging,
-)
+from optimum.utils import CONFIG_NAME, logging
 
+from ..test_utils import require_hf_token
 from .inference_utils import (
     MODEL_NAMES,
     SEED,
@@ -68,6 +66,7 @@ logger = logging.get_logger()
 
 
 @is_inferentia_test
+@require_hf_token
 class NeuronModelIntegrationTest(NeuronModelIntegrationTestMixin):
     MODEL_ID = MODEL_NAMES["bert"]
     if is_neuron_available():

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -13,8 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Tests for utility functions and classes."""
-import os
-import unittest
 
 from transformers import BertConfig, BertForSequenceClassification, PreTrainedModel, Wav2Vec2Config, Wav2Vec2Model
 
@@ -67,14 +65,3 @@ def test_patch_model():
         # Checking that the context manager exists.
         with wav2vec2_model.no_sync():
             pass
-
-
-def require_hf_token(test_case):
-    """
-    Decorator marking a test that requires huggingface hub token.
-    """
-    use_auth_token = os.environ.get("HF_TOKEN", None)
-    if use_auth_token is None:
-        return unittest.skip("test requires hf token as `HF_AUTH_TOKEN` environment variable")(test_case)
-    else:
-        return test_case

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -13,7 +13,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Tests for utility functions and classes."""
-
+import os
+import unittest
 
 from transformers import BertConfig, BertForSequenceClassification, PreTrainedModel, Wav2Vec2Config, Wav2Vec2Model
 
@@ -66,3 +67,14 @@ def test_patch_model():
         # Checking that the context manager exists.
         with wav2vec2_model.no_sync():
             pass
+
+
+def require_hf_token(test_case):
+    """
+    Decorator marking a test that requires huggingface hub token.
+    """
+    use_auth_token = os.environ.get("HF_TOKEN", None)
+    if use_auth_token is None:
+        return unittest.skip("test requires hf token as `HF_AUTH_TOKEN` environment variable")(test_case)
+    else:
+        return test_case

